Cargo.toml

```diff
@@ -1,5 +1,7 @@
 [workspace]
 members = [
+  "crates/*",
+
```

Comment:
I find https://matklad.github.io/2021/08/22/large-rust-workspaces.html a good structure, which broadly is consistent with the version in this PR.

Putting `bindings` separately is different from that advice. Though I can see the split making sense — I net to +/-0...

Reply:
A `crates` directory containing directories named after the crates is a well-established convention that helps new contributors quickly find the source code of crates. Putting the AST and the parser under compiler doesn't make any sense to me since they can very well be used on their own.

Anyway please open new issues for new discussions rather than making comments on old PRs.